src/Features/LanguageServer/Protocol/Handler/Initialize/InitializeHandler.cs

```diff
@@ -0,0 +1,41 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Composition;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.VisualStudio.LanguageServer.Protocol;
+
+namespace Microsoft.CodeAnalysis.LanguageServer.Handler
+{
+    [Shared]
+    [ExportLspMethod(Methods.InitializeName)]
+    internal class InitializeHandler : IRequestHandler<InitializeParams, InitializeResult>
+    {
+        public Task<InitializeResult> HandleRequestAsync(Solution solution, InitializeParams request, ClientCapabilities clientCapabilities, CancellationToken cancellationToken)
+        {
+            var result = new InitializeResult
```

Comment:
this could be single ton?

Reply:
I know that you are just porting live share one. so I am fine that you make work item and move on rather than change it in the PR.

Reply:
I can easily fix these ones now, so I might as well, keep these coming :)

If I can't easily fix it I'll add a work item.